core/src/main/java/org/elasticsearch/plugins/PluginManagerCliParser.java

```diff
@@ -0,0 +1,239 @@
+            }
+
+            return new ListPlugins(terminal, outputMode);
+        }
+
+        ListPlugins(Terminal terminal, OutputMode outputMode) {
+            super(terminal);
+            this.outputMode = outputMode;
+        }
+
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+            PluginManager pluginManager = new PluginManager(env, null, outputMode, DEFAULT_TIMEOUT);
+            try {
+                pluginManager.listInstalledPlugins(terminal);
+                return ExitStatus.OK;
+            } catch (Throwable e) {
+                terminal.printError("Error while listing plugins, reason: %s: %s", e.getClass().getSimpleName(), e.getMessage());
+                return ExitStatus.CODE_ERROR;
+            }
+        }
+    }
+
+    /**
+     * Remove a plugin
+     */
+    static class Remove extends CliTool.Command {
+
+        private static final String NAME = "remove";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, Remove.class).build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            String[] args = cli.getArgs();
+            if (args.length == 0) {
+                return exitCmd(ExitStatus.USAGE, terminal, "plugin name is missing (type -h for help)");
+            }
+
+            OutputMode outputMode = OutputMode.DEFAULT;
+            if (cli.hasOption("s")) {
+                outputMode = OutputMode.SILENT;
+            }
+            if (cli.hasOption("v")) {
+                outputMode = OutputMode.VERBOSE;
+            }
+
+            return new Remove(terminal, outputMode, args[0]);
+        }
+
+        private OutputMode outputMode;
+        final String pluginName;
+
+        Remove(Terminal terminal, OutputMode outputMode, String pluginToRemove) {
+            super(terminal);
+            this.outputMode = outputMode;
+            this.pluginName = pluginToRemove;
+        }
+
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+
+            PluginManager pluginManager = new PluginManager(env, null, outputMode, DEFAULT_TIMEOUT);
+            try {
+                terminal.println("-> Removing " + Strings.nullToEmpty(pluginName) + "...");
+                pluginManager.removePlugin(pluginName, terminal);
+                return ExitStatus.OK;
+            } catch (IllegalArgumentException e) {
+                terminal.printError("Failed to remove " + pluginName + ", reason: " + e.getMessage());
+                return ExitStatus.USAGE;
+            } catch (IOException e) {
+                terminal.printError("Failed to remove " + pluginName + ", reason: " + e.getMessage());
+                return ExitStatus.IO_ERROR;
+            } catch (Throwable e) {
+                terminal.printError("Error while removing plugins, reason: %s: %s", e.getClass().getSimpleName(), e.getMessage());
```

Comment:
If this is a code error, why not let the exception propagate and get the full stacktrace?